src/pages/signin/SignInPage.js

```diff
@@ -81,21 +82,22 @@ class SignInPage extends Component {
         const welcomeText = this.props.translate(`welcomeText.${showPasswordForm ? 'phrase4' : 'phrase1'}`);
 
         return (
-            <>
-                <SafeAreaView style={[styles.signInPage]}>
-                    <SignInPageLayout
-                        welcomeText={welcomeText}
-                        shouldShowWelcomeText={showLoginForm || showPasswordForm || !showResendValidationLinkForm}
-                        shouldShowWelcomeScreenshot={showLoginForm}
-                    >
-                        {showLoginForm && <LoginForm />}
-
-                        {showPasswordForm && <PasswordForm />}
-
-                        {showResendValidationLinkForm && <ResendValidationForm />}
-                    </SignInPageLayout>
-                </SafeAreaView>
-            </>
+            <SafeAreaView style={[styles.signInPage]}>
+                <SignInPageLayout
+                    welcomeText={welcomeText}
+                    shouldShowWelcomeText={showLoginForm || showPasswordForm || !showResendValidationLinkForm}
+                    shouldShowWelcomeScreenshot={showLoginForm}
+                >
+                    {/* Both LoginForm and PasswordForm should be present in the UI for Password Managers to work. */}
+                    <CollapsibleView isVisible={showLoginForm}>
+                        {isVisible => <LoginForm isVisible={isVisible} />}
+                    </CollapsibleView>
+                    <CollapsibleView isVisible={showPasswordForm}>
+                        {isVisible => <PasswordForm isVisible={isVisible} />}
+                    </CollapsibleView>
```

Comment:
After seeing this I think this utility would be a bit cleaner as an HOC. Passing the prop only to immediately consume it looks kind of funny.

Reply:
Hmm. Let me do it that way.

Reply:
Actually, prop is used to manage side effects. Ased on visibility.

Reply:
Could maybe pass the prop through to the wrapped component?